zipkin-server/src/main/java/zipkin2/server/internal/elasticsearch/ZipkinElasticsearchStorageAutoConfiguration.java

```diff
@@ -59,126 +48,116 @@
+    hostsSupplier.ifPresent(result::hostsSupplier);
     return result.build();
   }
 
-  // our elasticsearch impl uses an instance of OkHttpClient, not Call.Factory, so we have to trace
-  // carefully the pieces inside OkHttpClient
-  @Configuration
-  @ConditionalOnSelfTracing
-  static class TracingOkHttpClientBuilderEnhancer implements BeanPostProcessor {
-
-    @Autowired(required = false) HttpTracing httpTracing;
-    @Autowired(required = false) Tracing tracing;
-
-    @Override public Object postProcessBeforeInitialization(Object bean, String beanName) {
-      return bean;
-    }
-
-    @Override public Object postProcessAfterInitialization(Object bean, String beanName) {
-      if (httpTracing == null || !"zipkinElasticsearchHttpBuilder".equals(beanName)) return bean;
-      Tracer tracer = tracing.tracer();
-
-      OkHttpClient.Builder builder = (OkHttpClient.Builder) bean;
-      builder.addInterceptor(new Interceptor() {
-        /** create a local span with the same name as the request tag */
-        @Override public Response intercept(Chain chain) throws IOException {
-          // don't start new traces (to prevent amplifying writes to local storage)
-          if (tracer.currentSpan() == null) return chain.proceed(chain.request());
-
-          Request request = chain.request();
-          ScopedSpan span = tracer.startScopedSpan(request.tag().toString());
-          try {
-            return chain.proceed(request);
-          } catch (RuntimeException | IOException | Error e) {
-            span.error(e);
-            throw e;
-          } finally {
-            span.finish();
-          }
-        }
-      });
-      builder.addNetworkInterceptor(
-        TracingInterceptor.create(httpTracing.clientOf("elasticsearch")));
-      ExecutorService delegate = new Dispatcher().executorService();
-      builder.dispatcher(new Dispatcher(makeContextAware(delegate, tracing.currentTraceContext())));
-      return builder;
-    }
-  }
-
-  /**
-   * Decorates the input such that the {@link RequestContext#current() current request context} and
-   * the and the {@link CurrentTraceContext#get() current trace context} at assembly time is made
-   * current when task is executed.
-   */
-  static ExecutorService makeContextAware(ExecutorService delegate, CurrentTraceContext traceCtx) {
-    class TracingCurrentRequestContextExecutorService extends WrappingExecutorService {
-
-      @Override protected ExecutorService delegate() {
-        return delegate;
-      }
-
-      @Override protected <C> Callable<C> wrap(Callable<C> task) {
-        return RequestContext.current().makeContextAware(traceCtx.wrap(task));
-      }
-
-      @Override protected Runnable wrap(Runnable task) {
-        return RequestContext.current().makeContextAware(traceCtx.wrap(task));
-      }
+  @Bean @Qualifier(QUALIFIER) @ConditionalOnSelfTracing Consumer<HttpClientBuilder>
+  elasticsearchTracing(Optional<Tracing> tracing) {
```

Comment:
trying self-tracing seems to have a glitch as I'm only seeing spans for static assets.. looking into it now